File: lab1/profesor.cpp
#include "profesor.h"
#include "materie.h"

Profesor::Profesor(int id, string nume, string prenume, int anNastere)
   : Persoana(nume, prenume, anNastere)
{
   this->id = id;
}

void Profesor::adaugaNota(int materieId, string studentMatricol, string tipNota, float nota) {
   materii[materieId]->note[studentMatricol][tipNota] = nota;
}

float Profesor::calculeazaMedie(int materieId, string studentMatricol) {
   return materii[materieId]->medii[studentMatricol] = materii[materieId]->evaluare(studentMatricol);
}

File: lab1_java/src/main/java/com/lab1_java/Profesor.java
package com.lab1_java;

import java.util.HashMap;

public class Profesor extends Persoana {
    public int id;
    public HashMap<Integer, Materie> materii;
    
    Profesor(int id, String nume, String prenume, int anNastere) {
        super(nume, prenume, anNastere);
        this.materii = new HashMap<>();
        this.id = id;
    }
    
    public void adaugaNota(int materieId, String studentMatricol, String tipNota, float nota) {
        var notele = this.materii.get(materieId).note;
        if (!notele.containsKey(studentMatricol)) {
            notele.put(studentMatricol, new HashMap<>());
        }
        notele.get(studentMatricol).put(tipNota, nota);
    }
    
    public float calculeazaMedie(int materieId, String studentMatricol) {
        var newValue = this.materii.get(materieId).evaluare(studentMatricol);
        this.materii.get(materieId).medii.put(studentMatricol, newValue);
        return newValue;
    }
}
